contracts/oracle_aggregator/wrapper/ChainlinkSafeRateProviderCompositeWrapperWithThresholding.sol

```diff
@@ -0,0 +1,200 @@
+// SPDX-License-Identifier: MIT
+/* ———————————————————————————————————————————————————————————————————————————————— *
+ *    _____     ______   ______     __     __   __     __     ______   __  __       *
+ *   /\  __-.  /\__  _\ /\  == \   /\ \   /\ "-.\ \   /\ \   /\__  _\ /\ \_\ \      *
+ *   \ \ \/\ \ \/_\/_/  \ \  __<   \ \ \  \ \ \-.  \  \ \ \  \/_/\ \/ \ \____ \     *
+ *    \ \____-    /\_\   \ \_\ \_\  \ \_\  \ \_\"\_\  \ \_\    \ \_\  \/\_____\    *
+ *     \/____/     \/_/    \/_/ /_/   \/_/   \/_/ \/_/   \/_/     \/_/   \/_____/    *
+ *                                                                                  *
+ * ————————————————————————————————— dtrinity.org ————————————————————————————————— *
+ *                                                                                  *
+ *                                         ▲                                        *
+ *                                        ▲ ▲                                       *
+ *                                                                                  *
+ * ———————————————————————————————————————————————————————————————————————————————— *
+ * dTRINITY Protocol: https://github.com/dtrinity                                   *
+ * ———————————————————————————————————————————————————————————————————————————————— */
+
+pragma solidity ^0.8.20;
+
+import "../interface/chainlink/BaseChainlinkWrapper.sol";
+import "./ThresholdingUtils.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import {IPriceFeed} from "../interface/chainlink/IPriceFeed.sol";
+import {IRateProvider} from "../interface/IRateProvider.sol";
+import {IRateProviderSafe} from "../interface/IRateProviderSafe.sol";
+
+// Rate provider interface moved to interface/IRateProvider.sol
+
+/**
+ * @title ChainlinkRateProviderCompositeWrapperWithThresholding
+ * @notice Composes a Chainlink price feed with a generic rate provider, with optional thresholding
+ *         on each leg. Resulting price is scaled to BASE_CURRENCY_UNIT.
+ *
+ * Example for wstkscUSD -> scUSD:
+ *  - feed1: Chainlink EACAggregatorProxy for wstkscUSD -> stkscUSD (8 decimals typical)
+ *  - rateProvider: AccountantWithFixedRate (stkscUSD -> scUSD) returning a rate with `rateProviderUnit` decimals
+ *  - price(asset) = CL(wstkscUSD/stkscUSD in base) * RP(stkscUSD/scUSD in base) / BASE_CURRENCY_UNIT
+ */
+contract ChainlinkSafeRateProviderCompositeWrapperWithThresholding is
```

Comment:
Perhaps more obvious name here would be `ChainlinkERC4626RateCompositeWrapperWithThresholding` (other one can be shortened to `ERC4626RateWrapperWithThresholding`)

Since "SafeRateProvider" is ambiguous and obscures what the rate is actually communicating